edge-util/src/Microsoft.Azure.Devices.Edge.Util/azureLogAnalytics/AzureLogAnalytics.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util.AzureLogAnalytics
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Security.Cryptography;
+    using System.Text;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    /* Sample code from:
+    /* https://github.com/veyalla/MetricsCollector/blob/master/modules/MetricsCollector/AzureLogAnalytics.cs
+    /* https://dejanstojanovic.net/aspnet/2018/february/send-data-to-azure-log-analytics-from-c-code/
+    */
+
+    public sealed class AzureLogAnalytics
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<AzureLogAnalytics>();
+        static AzureLogAnalytics instance = null;
+        public string WorkspaceId { get; }
+        public string SharedKey { get; }
+        public string ApiVersion { get; }
+        AzureLogAnalytics(string workspaceId, string sharedKey, string apiVersion = "2016-04-01")
+        {
+            this.WorkspaceId = workspaceId;
+            this.SharedKey = sharedKey;
+            this.ApiVersion = apiVersion;
+        }
+
+        public static AzureLogAnalytics GetInstance()
+        {
+            Preconditions.CheckNotNull(instance);
+            return instance;
+        }
+
+        public static AzureLogAnalytics InitInstance(string workspaceId, string sharedKey)
+        {
+            if (instance == null)
+            {
+                instance = new AzureLogAnalytics(workspaceId, sharedKey);
+            }
+
+            return instance;
+        }
+
+        public async void PostAsync(string content, string LogType)
+        {
+            string dateString = DateTime.UtcNow.ToString("r");
+            Uri requestUri = new Uri($"https://{this.WorkspaceId}.ods.opinsights.azure.com/api/logs?api-version={this.ApiVersion}");
+            string signature = this.GetSignature("POST", content.Length, "application/json", dateString, "/api/logs");
+
+            var client = new HttpClient();
+            client.DefaultRequestHeaders.Add("Authorization", signature);
+            client.DefaultRequestHeaders.Add("Accept", "application/json");
+            client.DefaultRequestHeaders.Add("Log-Type", LogType);
+            client.DefaultRequestHeaders.Add("x-ms-date", dateString);
+
+            var contentMsg = new StringContent(content, Encoding.UTF8);
+            contentMsg.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+            Log.LogDebug(
+                client.DefaultRequestHeaders.ToString() +
+                contentMsg.Headers +
+                contentMsg.ReadAsStringAsync().Result);
+
+            var response = await client.PostAsync(requestUri, contentMsg).ConfigureAwait(false);
```

Comment:
it is possible that this method may throw exception, e.g. HttpRequestException.

Reply:
It depending on how mess up the request is. If say... you request is 404, it will be handle. 
but if your Http header is super mess up then it would stop the client.  Just to be super safe, I'm going to add a try-catch here.